Dave update getting started docs, and other supplimentary docs

davekaj opened the pull request:
Looking to just get some input on if there is anything I have missed. Please also check out for typos, bad grammar, or repetitiveness. 

I will be testing these on dharma this week, so I will be able to catch any errors as I go through the docs.

Resolves #388

nenadjaja commented on docs/getting-started.md:
```diff
@@ -34,221 +37,357 @@ The high level data flow is as follows:
+
+#### 4.2.3 Setting up your Ethereum Endpoint
+
+When you deploy your subgraph, you need to tell the subgraph where to start grabbing Ethereum data from. There are three typical ways you can grab data from Ethereum which the graph supports, which are Infura, local nodes, and ganache. 
```
> which the graph supports, which are Infura, local nodes, and ganache.

which The Graph supports, which are Infura, local nodes, and Ganache.

nenadjaja commented on docs/getting-started.md:
```diff
@@ -34,221 +37,357 @@ The high level data flow is as follows:
+
+When you deploy your subgraph, you need to tell the subgraph where to start grabbing Ethereum data from. There are three typical ways you can grab data from Ethereum which the graph supports, which are Infura, local nodes, and ganache. 
+
+Note, the command you must run requires that you have `rust` and `cargo` installed.
```
I wouldn't use "must" and I would explain what's the command for (e.g. The command for deploying your subgraph to The Graph assumes that you have `rust` and `cargo` installed... )

nenadjaja commented on docs/getting-started.md:
```diff
@@ -34,221 +37,357 @@ The high level data flow is as follows:
+
+##### 4.2.3.1 Infura
+
+[Infura](https://infura.io/) is supported for both websocket and http connections. This is the simplest way to source mainnet or testnet data, as you don't have to set up your own geth or parity node. However, it does sync slower than being connected to your own node. The following commands are passed to the `graph` command to indicate you want to use infura:
```
> The following commands

The following flags

nenadjaja commented on docs/getting-started.md:
````diff
@@ -34,221 +37,357 @@ The high level data flow is as follows:
+*Web Socket*
+```
+cargo run -p graph-node --release -- \
+  --postgres-url postgresql://<USERNAME><:PASSWORD>@localhost:5432/<YOUR_POSTGRES_DB_NAME> \
````
PASSWORD is optional, and I am not sure how to indicate that. We had it as [:PASSWORD], but we didn't use <> :) 
Also, I am still not sure if we want to use <YOUR_POSTGRES_DB_NAME> or just <POSTGRES_DB_NAME> and <YOUR_SUBGRAPH_NAME> or <SUBGRAPH_NAME>. Maybe we should have others weigh in on what's our convention, and then do it consistently. Just a thought

davekaj replied:
probably better without YOUR. we should get a convention though

passowrd is optional in some cases i believe. one user opened an issue, apparently he needed the password for linux. or else if you setup your postgres to be password protected it might want it. it seems like it can go either way

nenadjaja commented on docs/getting-started.md:
```diff
@@ -34,221 +37,357 @@ The high level data flow is as follows:
+Also, if you set up `graph deploy` as outlined in the [graph-cli](https://github.com/graphprotocol/graph-cli) repository, your graph node should automatically start tracking any new `subgraph ids` created from changes uploaded from `yarn build-ipfs`. 
+
+## 7 Example Subgraphs You can look to 
+
```
This is great! Do we want to add `memefactory` as well?

davekaj replied:
added

Jannis commented:
I believe we've been using `SUBGRAPH_NAME` for mandatory values and `[PORT]` for optional values, although notations here vary. `commander`, which we are using for `graph-cli`, uses `<SUBGRAPH_NAME>` and `[PORT]`, so I'm happy to follow that.

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +37,359 @@ The high level data flow is as follows:
+2. Build and Deploy The Subgraph
+    1. [Build and deploy your mappings to IPFS](#41-build-and-deploy-your-mappings-to-ipfs)
+    
+    1. [Deploy your subgraph to your local Graph Node](#42-deploy-your-subgraph-to-your-local-graph-node)
```
Can we merge the `build` and `deploy` items into a single one? That's about starting the node and running `yarn deploy`?

Jannis commented on docs/getting-started.md:
````diff
@@ -34,221 +37,359 @@ The high level data flow is as follows:
-}
-```
+## 3 Defining your Subgraph 
+In The Graph, we refer to your project's GraphQL endpoint as a *subgraph*, because once deployed to the decentralized network, it will be just one subset of a global GraphQL endpoint.
````
It will not "be just one subset of a global GraphQL endpoint". That's sounds insignificant.

`It will form a part of a global graph of blockchain data.`?

Jannis replied:
We actually refer to the entire project as the "subgraph", not just to the GraphQL endpoint.

Jannis commented on docs/getting-started.md:
````diff
@@ -34,221 +37,359 @@ The high level data flow is as follows:
-  }
-}
-```
+### 3.1 Define Your Source Data and Create a Manifest
````
We call these data sources, not source data.

nenadjaja commented on docs/getting-started.md:
```diff
@@ -1,28 +1,31 @@
 > **Note**:  this project is heavily WIP and until it reaches v1.0 the API is subject to change in breaking ways without notice.
 
-## 1 Introduction
+## 0 Introduction
```
I like the `0` cause there are no actual steps involved to get it running, just reading through the theory. Then the work starts at step `1`

davekaj replied:
I'm okay keeping it with 0 for now, because i think this whole doc can be designed a lot better. Having the whole entire docs in the readme makes a long doc. maybe we sepearte section 0 into another file next week

davekaj commented on docs/getting-started.md:
```diff
@@ -34,221 +37,359 @@ The high level data flow is as follows:
+    4. [Generate types for your mapping with the Graph-CLI](#33-generate-types-for-your-mapping-with-the-graph-cli)
+    5. [Write your mappings](#34-write-your-mappings)
+2. Build and Deploy The Subgraph
+    1. [Build and deploy your mappings to IPFS](#41-build-and-deploy-your-mappings-to-ipfs)
```
updated 95% of them . about 5% seemed okay

davekaj commented:
Addressed all the comments. 

I also think jannis got half way through, so there might be some more mistakes. 

I will be back on tonight and can address all fixes and get this merged tonight!

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +37,379 @@ The high level data flow is as follows:
+
+Use Metamask --> import account ---> private key.
+
+You now have an Ethereum account with a ton of ether, and you should be able to set up the migrations on this network and use Truffle. Now, send some ether to the previous account that got created and unlocked, this way you can run `truffle migrations` with this account. Weirdly,  `0x00a329c0648769A73afAc7F9381E08FB43dBEA72` does not get unlocked by parity, probably because it's password is an em ...
```
`ether` -> `Ether`?
`truffle migrations` -> `truffle migrate`
`it's password` -> `its password`
`either` -> `Ether`

Jannis replied:
`from the default account` -> `from the default account to this new account`

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +37,379 @@ The high level data flow is as follows:
+
+You now have an Ethereum account with a ton of ether, and you should be able to set up the migrations on this network and use Truffle. Now, send some ether to the previous account that got created and unlocked, this way you can run `truffle migrations` with this account. Weirdly,  `0x00a329c0648769A73afAc7F9381E08FB43dBEA72` does not get unlocked by parity, probably because it's password is an em ...
+
+#### 4.2.4 Syncing with the Testnet
```
`the Testnet` -> `a Public Testnet`

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +37,379 @@ The high level data flow is as follows:
+
+#### 4.2.4 Syncing with the Testnet
+
+If you want to sync with the testnet instead of the mainnet, just make sure the local node is a testnet node, or that you are hitting the Infura testnet endpoints. 
```
`with the testnet` -> `using a public testnet such as Kovan, Rinkeby or Ropsten`

Jannis replied:
`endpoints` -> `endpoint`

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +37,379 @@ The high level data flow is as follows:
+If you want to sync with the testnet instead of the mainnet, just make sure the local node is a testnet node, or that you are hitting the Infura testnet endpoints. 
+
+## 5 Query the local Graph Node
+With the subgraph deployed to the locally running Graph Node, visit http://127.0.0.1:8000/ to open up a [Graphiql](https://github.com/graphql/graphiql) interface where you can explore the deployed GraphQL API for the subgraph by issuing queries and viewing the schema. 
```
`Graphiql` -> `GraphiQL`

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +37,379 @@ The high level data flow is as follows:
 
-## 6 Buidl 🚀
-Start building world-changing dApps on top of your newly deployed GraphQL interface 🗿✨.
+You can also organize the queries. This query below would organize all tokens by their ID, and return the current owner of each token. 
```
I find `organize` an odd term here. To me it suggests multiple queries that are somehow organized into e.g. categories.

`organize the queries` -> `sort, filter or paginate query results`?

Jannis commented on docs/getting-started.md:
````diff
@@ -34,221 +37,379 @@ The high level data flow is as follows:
+}
+```
+
+GraphQL has a ton of functionality! Once again, check out the [Query API](graphql-api.md#1-queries) to figure out how to use all query functionalities. 
````
`has` -> `provides`

Jannis replied:
`query functionalities` -> `query features supported`

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +37,379 @@ The high level data flow is as follows:
+
+## 6 Changing the schema, mappings, and manifest, and launching a new Subgraph
+
+When you first start building the subgraph, it is likely you will make a few iterations to the manifest, mappings, or schema. If you update any of them, you should run `yarn codegen` and `yarn deploy`. This will post the new files on ipfs and deploy the new subgraph. Note that the graph node can track multiple subgraphs, so you can do this as many times as you like.
```
`you should run` -> `you should rerun`

Jannis replied:
`ipfs` -> `IPFS`
`deploy the new subgraph` -> `update the subgraph deployment`
`graph node` -> `Graph Node`

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +37,379 @@ The high level data flow is as follows:
+
+When you first start building the subgraph, it is likely you will make a few iterations to the manifest, mappings, or schema. If you update any of them, you should run `yarn codegen` and `yarn deploy`. This will post the new files on ipfs and deploy the new subgraph. Note that the graph node can track multiple subgraphs, so you can do this as many times as you like.
+
+## 7 Example Subgraphs You can look to 
```
Drop the `You can look to`

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +37,379 @@ The high level data flow is as follows:
+* https://github.com/graphprotocol/adchain-subgraph
+* https://github.com/graphprotocol/memefactory-subgraph (only for testing, not live on Ethereum mainnet)
+
+It might be worth your while to clone one of them, and start the project from there. 
```
Not a good idea, it could lead to disqualification at hackathons, because hackers are only allowed to start coding after the hackathon has kicked off. Cloning existing repos will lead to commits that predate the event.

davekaj replied:
yeah I was writing it with the thought of any person coming here, and not just the hackathon. I will just mention that these are our repositories they can refer to

Jannis commented on docs/graphql-api.md:
```diff
@@ -1,11 +1,11 @@
 # 1 Queries
 # 1.1 Basics
-For each type `Entity` which you define in your schema, an `entity` and `entities` field will be generated on the top-level `Query` type.
+For each type `Entity` which you define in your schema, an `entity` and `entities` field will be generated on the top-level `Query` type. Note that `query` does not need to be included at the top of the `graphql` query when using The Graph. 
```
`which you define` -> `that you define`

Jannis replied:
Drop the `Now, an `entity` is what you will define in order to detemine how the `data source` will be indexed in a Graph Node.`. That's just how GraphQL works, nothing specific here. There's a typo in the sentence anyway (`detemine`).

davekaj replied:
looks like the 2nd comment got copy pasted from above, that sentance was in a different file. I fixed the 1st one though

davekaj commented on README.md:
```diff
@@ -33,24 +33,27 @@ For Ethereum network data you can either run a local node or use Infura.io:
-Try your OS username as `USERNAME` and `PASSWORD` is optional (it can be anything).
+Try your OS username as `USERNAME` and `PASSWORD`. The password might be optional, it depends on your setup. 
 
 This will also spin up a GraphiQL interface at `http://127.0.0.1:8000/`.
```
added

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +36,368 @@ The high level data flow is as follows:
-What this command does is it looks at the contract ABIs defined in your subgraph manifest, and for each datasource it generates TypeScript types (actually AssemblyScript types, but more on that later) for the smart contracts your mappings script will interface with, including the types of public methods and events.
+
+### 3.3 Graph CLI
+Once you have the `subgraph.yaml` manifest and the `./schema.graphql` file, you are ready to use the Graph CLI to set up the subgraph directory. The GRAPH CLI is a command line tool that contains helpful commands for deploying the subgraphs. Before continuing with this guide, please go to the [Graph CLI README] (https://github.com/graphprotocol/graph-cli/) and follow the instructions up to Step 7 ...
```
Broken link: Remove the space between [...] and (...)

Jannis commented on docs/getting-started.md:
```diff
@@ -34,221 +36,368 @@ The high level data flow is as follows:
+### 3.3 Graph CLI
+Once you have the `subgraph.yaml` manifest and the `./schema.graphql` file, you are ready to use the Graph CLI to set up the subgraph directory. The GRAPH CLI is a command line tool that contains helpful commands for deploying the subgraphs. Before continuing with this guide, please go to the [Graph CLI README] (https://github.com/graphprotocol/graph-cli/) and follow the instructions up to Step 7 ...
+
+Once you run `yarn codegen` as outlined in the GRAPH CLI README, you are ready to create mappings! What this command does is it looks at the contract ABIs defined in the subgraph manifest, and generates TypeScript types for all of the ABIs (actually AssemblyScript types, but more on that later) for the smart contracts the mappings script will interface with, including the types of public methods  ...
```
`GRAPH CLI` -> `Graph CLI`
